src/tasks.ts

```diff
@@ -329,42 +329,592 @@ export async function callFirestore(
     throw err;
   }
 }
+/**
+ * Create a Firebase Auth user
+ * @param adminInstance - Admin SDK instance
+ * @param properties - The properties to set on the new user record to be created
+ * @param tenantId - Optional ID of tenant used for multi-tenancy
+ * @returns Promise which resolves with a UserRecord
+ */
+export function createAuthUser(
+  adminInstance: any,
+  properties: auth.CreateRequest,
+  tenantId?: string,
+): Promise<auth.UserRecord | null> {
+  return getAuth(adminInstance, tenantId)
+    .createUser(properties)
+    .catch((err) => {
+      if (err.code === 'auth/email-already-exists') return null;
```

Comment:
Good point.
I think at the time I thought to do something along the lines what happens with the login command, where it is idempotent and skips trying to log in when it already is.
But when I look at the login command it indeed does cy.log when the user already is logged in.
The cy object isn't as easily accessible within tasks though. Would it be better to remove the catch() in the function here and move it to the command instead, where cy.log can be added?
The errors could also just not be silenced at all, I'm fine with it either way.

Reply:
Yeah I think we would want it in the command instead - that said, we will have to check on how the error is parsed since the task/command boundary goes from node into browser through Cypress, so parsing sometimes gets interesting

Reply:
A problem with trying to do error handling in the commands, is that Cypress doesn't really want you to. I have some sort of workaround but I'm net yet sure if I like it really.
Currently I make the task return a string with the same error message. Since the tasks that have these errors we might want to suppress all return some sort of object, there is no collision with the (literal) strings from the error messages. The command using the task can then 'catch' the errors by checking for those literal strings.
In the case of `createAuthUser` I made it such that when a user with the email/phoneNumber already exists, it tries to find the user to return the uid of. This way it has a similar type of idempotency as login.
The `getAuthUser[By…]` commands return null when the specified user wasn't found, similar to how a read from a rtdb path returns null when nothing exists at that place. The need for cy.log is less here, since the user explicitly gets a null returned and thus already 'knows' the user does not exist.

Let me know your thoughts!

PS: I also tried to add more specific typings to the cypress chainables, not sure if this is useful, but if it impedes something I am not aware of, just the single commit would have to be reverted.

Reply:
Yeah I like what you are saying about the idempotency matching with login

Also agree that the getAuthUser commands returning null makes sense to match RTDB/Firestore commands